core/src/main/scala/com/github/gvolpe/fs2redis/connection/Fs2RedisClusterClient.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Fs2 Redis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.gvolpe.fs2redis.connection
+
+import cats.implicits._
+import cats.effect.{ Concurrent, Resource, Sync }
+import com.github.gvolpe.fs2redis.domain.{ DefaultRedisClusterClient, Fs2RedisClusterClient }
+import com.github.gvolpe.fs2redis.effect.{ JRFuture, Log }
+import io.lettuce.core.RedisURI
+import io.lettuce.core.cluster.RedisClusterClient
+
+import scala.collection.JavaConverters._
+
+object Fs2RedisClusterClient {
+
+  private[fs2redis] def acquireAndRelease[F[_]: Concurrent: Log](
+      uri: RedisURI*
+  )(implicit F: Sync[F]): (F[Fs2RedisClusterClient], Fs2RedisClusterClient => F[Unit]) = {
+
+    val acquire: F[Fs2RedisClusterClient] =
+      Log[F].info(s"Acquire Redis Cluster client") *>
+        Sync[F].delay {
+          val client = RedisClusterClient.create(uri.asJava)
+          client.getPartitions
```

Comment:
Basically it retrieves the cluster view to compute correct hash slots (and corresponding nodes on the driver level for operations (set/get/etc). 

The problem here is that in `com.github.gvolpe.fs2redis.interpreter.Fs2Redis#acquireAndReleaseCluster` I use `client.underlying.connectAsync` that requires pre-loaded partitions via `client.getPartitions` call (see `io.lettuce.core.cluster.RedisClusterClient#connectClusterAsync`):

```java
if (partitions == null) {
  return Futures.failed(new IllegalStateException(
               "Partitions not initialized. Initialize via RedisClusterClient.getPartitions()."));
}
```

For sync connections they do it for you - `io.lettuce.core.cluster.RedisClusterClient#connect(io.lettuce.core.codec.RedisCodec<K,V>)`

Reply:
Totally agree about naming 😀 
Take a look, I've extracted that part into separate method, but honestly I don't like it so much because  `initializeClusterPartitions` mutates `client` argument and we have to pass that mutable state further.